code/_onclick/hud/human.dm

```diff
@@ -42,46 +41,93 @@
 			inv_box.set_dir(slot_data["dir"])
 
 		if(slot_data["toggle"])
-			src.other += inv_box
+			toggleable_inventory += inv_box
 			has_hidden_gear = 1
 		else
-			src.adding += inv_box
+			static_inventory += inv_box
 
 	if(has_hidden_gear)
-		using = new /obj/screen()
-		using.SetName("toggle")
+		using = new /obj/screen/toggle_inv()
 		using.icon = ui_style
-		using.icon_state = "other"
-		using.screen_loc = ui_inventory
 		using.color = ui_color
 		using.alpha = ui_alpha
-		src.adding += using
+		static_inventory += using
 
 	// Draw the attack intent dialogue.
 	if(hud_data.has_a_intent)
 
-		using = new /obj/screen/intent()
-		src.adding += using
-		action_intent = using
-
-		hud_elements |= using
+		action_intent = new /obj/screen/intent(owner)
+		static_inventory += action_intent
 
 	if(hud_data.has_healthbar)
 
-		using = new /obj/screen/meter/health(target.client)
-		src.adding += using
-		hud_elements |= using
+		hud_healthbar = new /obj/screen/meter/health(owner)
+		hud_healthbar.L = owner
+		infodisplay += hud_healthbar
+
+		remaining_meter = new /obj/screen/meter_component/current(hud_healthbar)
```

Comment:
this stuff definitely doesn't belong here, the meter creates its own components

Reply:
Updated